src/api/procedures/createSecurityToken.ts

```diff
@@ -118,16 +121,26 @@ export async function prepareCreateSecurityToken(
     fee = new BigNumber(0);
   }
 
-  this.addTransaction(
-    tx.asset.createAsset,
-    { fee },
-    rawName,
-    rawTicker,
-    rawIsDivisible,
-    rawType,
-    rawIdentifiers,
-    rawFundingRound
-  );
+  if (!totalSupply || (totalSupply && totalSupply.eq(0))) {
+    this.addTransaction(
+      tx.asset.createAsset,
+      { fee },
+      rawName,
+      rawTicker,
+      rawIsDivisible,
+      rawType,
+      rawIdentifiers,
+      rawFundingRound
+    );
+  } else {
+    this.addBatchTransaction(tx.asset.createAsset, { fee }, [
+      tuple(rawName, rawTicker, rawIsDivisible, rawType, rawIdentifiers, rawFundingRound),
+    ]);
+
+    const rawTotalSupply = numberToBalance(totalSupply, context, isDivisible);
+
+    this.addBatchTransaction(tx.asset.issue, {}, [tuple(rawTicker, rawTotalSupply)]);
```

Comment:
I forgot that batching different kinds of transactions isn't currently supported by the SDK. What you're doing here is adding two separate batches.

Until we support that, I suggest just using `addTransaction` here. You should refactor the code like so:

- create token
- if totalSupply exists and is non-zero, add an issue tx